lib/ansible/parsing/vault/__init__.py

```diff
@@ -927,7 +927,7 @@ def create_file(self, filename, secret, vault_id=None):
         """ create a new encrypted file """
 
         dirname = os.path.dirname(filename)
-        if not os.path.exists(dirname):
+        if dirname != '' and not os.path.exists(dirname):
```

Comment:
Can we do `if dirname and not os.path.exists(dirname):` instead? Looks nicer I think.